### PR TITLE
translate-c: correctly handle pointers to opaque demoted structs

### LINKED ISSUE #4686: translate-c: demotion to opaque vs C pointers cannot point opaque types
Trying to `@cImport` the arcan headers, I get:
```
error: C pointers cannot point opaque types
    addr: [*c]struct_arcan_shmif_page,
              ^
zig-cache/o/u4lVIl08Y59a_k_8fG9jGF26V9EEYInpaLPpcnjqEk53IdoYRCSC1DMWFJ7qka4D/cimport.zig:1503:5: note: referenced here
    addr: [*c]struct_arcan_shmif_page,
    ^
```

translate-c output snippet:
```zig
pub const struct_arcan_shmif_page = // /usr/include/arcan/shmif/arcan_shmif_control.h:943:34: warning: unsupported type: 'Atomic'
    @OpaqueType(); // /usr/include/arcan/shmif/arcan_shmif_control.h:905:8: warning: struct demoted to opaque type - unable to translate type of field abufused
const union_unnamed_29 = extern union {
    vidp: [*c]shmif_pixel,
    floatp: [*c]f32,
    vidb: [*c]u8,
};
const union_unnamed_31 = extern union {
    audp: [*c]shmif_asample,
    audb: [*c]u8,
};
pub const struct_arcan_shmif_region = extern struct {
    x1: u16,
    x2: u16,
    y1: u16,
    y2: u16,
};
pub const struct_shmif_hidden = @OpaqueType();
pub const struct_shmif_ext_hidden = @OpaqueType();
pub const struct_arcan_shmif_cont = extern struct {
    addr: [*c]struct_arcan_shmif_page,
    unnamed_28: union_unnamed_29,
    unnamed_30: union_unnamed_31,
    oflow_cookie: i16,
    abufused: u16,
    abufpos: u16,
    abufsize: u16,
    abufcount: u16,
    abuf_cnt: u8,
    epipe: file_handle,
    shmh: file_handle,
    shmsize: usize,
    vsem: sem_handle,
    asem: sem_handle,
    esem: sem_handle,
    w: usize,
    h: usize,
    stride: usize,
    pitch: usize,
    adata: u32,
    samplerate: usize,
    hints: u8,
    dirty: struct_arcan_shmif_region,
    cookie: u64,
    user: ?*c_void,
    priv: ?*struct_shmif_hidden,
    privext: ?*struct_shmif_ext_hidden,
    segment_token: u32,
    vbufsize: usize,
};
```

Related to #4248

### PR DESCRIPTION
**Pull request by Vexu**
Closes #4686

### CODE REVIEW

File: src/translate_c.zig [hunk 1]
```diff
@@ -4853,6 +4856,50 @@ fn transType(rp: RestorePoint, ty: *const clang.Type, source_loc: clang.SourceLo
     }
 }
 
+fn qualTypeWasDemotedToOpaque(c: *Context, qt: ZigClangQualType) bool {
```
Comment (LemonBoy) [inline src/translate_c.zig:4859]: Wouldn't it be better to memoize this info?

Comment (Vexu) [reply src/translate_c.zig:4859]: I doubt it'd make any noticeable difference. Most of the time this won't do more than a single switch and possibly a hashmap lookup which would happen anyways if memoized.

File: test/translate_c.zig [hunk 2]
```diff
@@ -3,6 +3,26 @@ const std = @import("std");
...
+    cases.add("pointer to opaque demoted struct",
+        \\typedef struct {
+        \\    _Atomic int foo;
+        \\} Foo;
+        \\
+        \\typedef struct {
+        \\    Foo *bar;
+        \\} Bar;
+    , &[_][]const u8{
+        \\pub const struct_unnamed_1 = //
```
Comment (daurnimator) [inline test/translate_c.zig:15]: I'm confused what's going on here

Comment (daurnimator) [reply test/translate_c.zig:15]: I see:
```
========= Expected to find: ===================
pub const struct_unnamed_1 = //
========= But file does not contain it: =======
const struct_unnamed_1 = // zig-cache\o\Tu5jQm49pp_tXOump2AzNz7zspEr-ur1UBaKvKvR87raZso0SY-uFdxe2QYQZ4SF\source.h:2:17: warning: unsupported type: 'Atomic'
    opaque {}; // zig-cache\o\Tu5jQm49pp_tXOump2AzNz7zspEr-ur1UBaKvKvR87raZso0SY-uFdxe2QYQZ4SF\source.h:1:9: warning: struct demoted to opaque type - unable to translate type of field foo
pub const Foo = struct_unnamed_1;
const struct_unnamed_2 = extern struct {
    bar: ?*Foo,
};
pub const Bar = struct_unnamed_2;
```

Comment (daurnimator) [reply test/translate_c.zig:15]: OH, is it just that the `pub` is in the wrong spot?

File: test/translate_c.zig [hunk 3]
```diff
@@ -3,6 +3,26 @@ const std = @import("std");
...
+        \\typedef struct {
+        \\    _Atomic int foo;
+        \\} Foo;
+        \\
+        \\typedef struct {
+        \\    Foo *bar;
+        \\} Bar;
+    , &[_][]const u8{
+        \\const struct_unnamed_1 = //
+    ,
```
Comment (daurnimator) [inline test/translate_c.zig:16]: comma placement is odd.

**Pull request merged**